### PR TITLE
Reconcile differences between frontend & backend default version behavior

### PR DESCRIPTION
**Pull request by bennerv**
### Which issue this PR addresses:

Fixes logical differences between frontend & backend handling the default version

### What this PR does / why we need it:

We need to support the default version no matter what.  This is because if a customer does not specify a version, we set the default version.  So we cannot disable it as it would provide a poor experience.  

### Test plan for issue:

Create a default version cluster when a record for it does not exist in cosmosdb.  

### Is there any documentation that needs to be updated for this PR?

Probably, but don't think we have existing documentation on this.

### CODE REVIEW

**Review by s-amann — DISMISSED** [review]

File: pkg/cluster/version.go [hunk 1]
```diff
@@ -40,34 +58,6 @@ func (service *openShiftClusterDocumentVersionerService) Get(ctx context.Context
...
 
-	errUnsupportedVersion := api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.clusterProfile.version", "The requested OpenShift version '%s' is not supported.", requestedInstallVersion)
-
-	// when we have no OpenShiftVersion entries in CosmoDB, default to building one using the InstallStream
-	if len(activeOpenShiftVersions) == 0 {
-		if requestedInstallVersion != version.InstallStream.Version.String() {
-			return nil, errUnsupportedVersion
-		}
-
-		installerPullSpec := env.LiveConfig().DefaultInstallerPullSpecOverride(ctx)
```
Comment (darthhexx) [inline pkg/cluster/version.go:51]: Noting that in this change we're losing the "LiveConfig" capability to override the default installer pullspec. We aren't using this feature currently, so its not breaking anything, but worth noting that we may want to reintroduce this when LiveConfig is put live.

File: pkg/cluster/version.go [hunk 2]
```diff
@@ -28,6 +28,24 @@ type openShiftClusterDocumentVersionerService struct{}
...
 	requestedInstallVersion := doc.OpenShiftCluster.Properties.ClusterProfile.Version
 
+	// add the default OCP version as we require it as an install target
+	// if this is removed, we need to also update the logic in
+	// pkg/frontend/frontend.go, pkg/frontend/validate.go
+	defaultVersion := &api.OpenShiftVersion{
+		Properties: api.OpenShiftVersionProperties{
+			Version:           version.InstallStream.Version.String(),
+			OpenShiftPullspec: version.InstallStream.PullSpec,
+			InstallerPullspec: fmt.Sprintf("%s/aro-installer:release-%s", env.ACRDomain(), version.InstallStream.Version.MinorVersion()),
```
Comment (darthhexx) [inline pkg/cluster/version.go:38]: Also wanted to highlight that by using this hard-coded InstallerPullspec, together with dropping the LiveConfig override, we are losing the ability to set a different value for the default.

Comment (bennerv) [reply pkg/cluster/version.go]: Fixed by adding the installer spec override at the top of the function.

**Review by petrkotas — APPROVED** [review]
Works for me. Thanks 

One thing that caught my eye is, that if we set the override wrong it is not validated. However, it is us who sets the override, so it is not a big issue.

**Pull request merged**